### PR TITLE
Equity baskets

### PR DESCRIPTION
**Pull request by tommyjcarpenter**
Still have to add some docs and still need to add the POST to /identifier/bloomberg - that can be a seperate PR. 

@ykifle  I would suggest we merge and I can add those two in another PR.

### CODE REVIEW

File: merqube_client_lib/templates/equity_baskets/util.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    load index specific info from config file + validate the data
+    """
+    index_info = cast(dict[str, Any], json.load(open(config_file_path, "r")))
+
+    for k in required_fields:
+        if k not in index_info:
+            raise ValueError(f"Missing key {k} in index info")
+
+    try:
+        for k in ["base_date"]:
```
Comment (ykifle) [inline merqube_client_lib/templates/equity_baskets/util.py:100]: Why loop?

File: merqube_client_lib/templates/equity_baskets/util.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    return client, template, index_info, inner_spec
+
+
+def configure_index(template: IndexDefinitionPost, index_info: dict[str, Any], inner_spec: dict[str, Any]) -> None:
+    """
+    set up the index from index info
+    """
+    for k in TOP_LEVEL:
+        setattr(template, k, index_info[k])
```
Comment (ykifle) [inline merqube_client_lib/templates/equity_baskets/util.py:138]: `base_date` format at top level is `%Y/%m/%d`. I'm not exactly sure if this matters. It depends on how the website uses it.

File: merqube_client_lib/templates/equity_baskets/util.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            pd.Timestamp(index_info[k])
+    except ValueError as e:
+        raise ValueError(
+            "{k} must be a valid date. If the index should run at a certain time of day UTC, that should be part of the ISO8601 formatted first_run_date"
+        ) from e
+
+    if (tz := index_info.get("timezone")) and tz not in pytz.all_timezones:
+        raise ValueError("Invalid timezone string: {tz}")
+
+    return index_info
```
Comment (ykifle) [inline merqube_client_lib/templates/equity_baskets/util.py:110]: Validate run_hour and run_minute

File: merqube_client_lib/templates/equity_baskets/util.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+    template.identifiers = (
+        [IdentifierUUIDRef(name=bbgt, provider=Provider.bloomberg)] if (bbgt := index_info.get("bbg_ticker")) else []
+    )
+
+    template.stage = Stage.prod
+    template.run_configuration.job_enabled = True
+
+    if index_info.get("is_intraday"):
+        assert template.intraday and template.intraday.publish_config and template.intraday.publish_config
```
Comment (ykifle) [inline merqube_client_lib/templates/equity_baskets/util.py:168]:
```suggestion
        assert template.intraday and template.intraday.publish_config
```

**Review by ykifle — APPROVED** [review]

File: merqube_client_lib/templates/equity_baskets/util.py (hunk 1 above)
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/util.py:100]: was originally a longer list - ill make this a DATE_KEYS variable

File: merqube_client_lib/templates/equity_baskets/util.py (hunk 2 above)
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/util.py]: done

File: merqube_client_lib/templates/equity_baskets/util.py (hunk 3 above)
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/util.py:117]: done

**Review by ykifle — APPROVED** [review]

File: merqube_client_lib/templates/equity_baskets/single_stock_total_return_corax/template.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        "set base_date to a business day before the intended start date for the real-time calcs",
+        "[OPTIONAL] bbg_ticker: due to the limitations of the Bloomberg ticker creation process (no API), this must be a pre-created ticker. You can email MerQube a list of tickers to create on your behalf at support@merqube.com. Then, you provide those as input to these indices.",
+        "[OPTIONAL] currency: if not provided, defaults to USD. Eg CAD, EUR, GBP, JPY, etc",
+        "set name to the name on Merqube's website for the index. Can also be used for the ticker on Bloomberg. This must be globally unique (you will get a 409 if the name is taken)",
+        "set namespace to your organization's namespace for indices",
+        "set run_hour and run_minute to the time of day that the index should be calculated, in the timezone specified. Eg 18 and 0 for 6pm",
+        "timezone is optional, defaulting to US/Eastern. Must be a valid pytz timezone",
+        "underlying RIC, the reuters ric of the stock the TR index is on top of"
+    ],
+    "apikey": "XXXX",
```
[the remark this answers is not in the document]
Comment (ykifle) [reply merqube_client_lib/templates/equity_baskets/single_stock_total_return_corax/template.json:12]: I'm all for it. I prefer that as well.

**Pull request merged**